openrtb_ext/bidders.go

```diff
@@ -115,116 +128,130 @@ const (
-	"yieldlab":          BidderYieldlab,
-	"yieldmo":           BidderYieldmo,
-	"yieldone":          BidderYieldone,
-	"zeroclickfraud":    BidderZeroClickFraud,
+// BidderNames returns all core bidders.
+func BidderNames() []BidderName {
+	return []BidderName{
+		Bidder33Across,
+		BidderAdform,
+		BidderAdgeneration,
+		BidderAdhese,
+		BidderAdkernel,
+		BidderAdkernelAdn,
+		BidderAdman,
+		BidderAdmixer,
+		BidderAdOcean,
+		BidderAdoppler,
+		BidderAdpone,
+		BidderAdprime,
+		BidderAdtarget,
+		BidderAdtelligent,
+		BidderAdvangelists,
+		BidderAJA,
+		BidderApplogy,
+		BidderAppnexus,
+		BidderAudienceNetwork,
+		BidderAvocet,
+		BidderBeachfront,
+		BidderBeintoo,
+		BidderBetween,
+		BidderBrightroll,
+		BidderColossus,
+		BidderConnectAd,
+		BidderConsumable,
+		BidderConversant,
+		BidderCpmstar,
+		BidderDatablocks,
+		BidderDmx,
+		BidderEmxDigital,
+		BidderEngageBDR,
+		BidderEPlanning,
+		BidderGamma,
+		BidderGamoshi,
+		BidderGrid,
+		BidderGumGum,
+		BidderImprovedigital,
+		BidderInMobi,
+		BidderInvibes,
+		BidderIx,
+		BidderKidoz,
+		BidderKrushmedia,
+		BidderKubient,
+		BidderLifestreet,
+		BidderLockerDome,
+		BidderLogicad,
+		BidderLunaMedia,
+		BidderMarsmedia,
+		BidderMgid,
+		BidderMobileFuse,
+		BidderNanoInteractive,
+		BidderNinthDecimal,
+		BidderOpenx,
+		BidderOrbidder,
+		BidderPubmatic,
+		BidderPubnative,
+		BidderPulsepoint,
+		BidderRhythmone,
+		BidderRTBHouse,
+		BidderRubicon,
+		BidderSharethrough,
+		BidderSmaato,
+		BidderSmartAdserver,
+		BidderSmartRTB,
+		BidderSmartyAds,
+		BidderSomoaudience,
+		BidderSonobi,
+		BidderSovrn,
+		BidderSynacormedia,
+		BidderTappx,
+		BidderTelaria,
+		BidderTriplelift,
+		BidderTripleliftNative,
+		BidderUcfunnel,
+		BidderUnruly,
+		BidderValueImpression,
+		BidderVerizonMedia,
+		BidderVisx,
+		BidderVrtcal,
+		BidderYeahmobi,
+		BidderYieldlab,
+		BidderYieldmo,
+		BidderYieldone,
+		BidderZeroClickFraud,
+	}
 }
 
-// BidderList returns the values of the BidderMap
-func BidderList() []BidderName {
-	bidders := make([]BidderName, 0, len(BidderMap))
-	for _, value := range BidderMap {
-		bidders = append(bidders, value)
+// BuildBidderNameLookup builds a map of string to BidderName, to remain compatbile with the
+// prebioud BidderMap variable.
+func BuildBidderNameLookup() map[string]BidderName {
+	lookup := make(map[string]BidderName)
+	for _, name := range BidderNames() {
+		lookup[string(name)] = name
 	}
-	return bidders
+	return lookup
 }
 
-func (name BidderName) MarshalJSON() ([]byte, error) {
-	return []byte(name), nil
+func BuildBidderNameHashSet() map[string]struct{} {
+	hashSet := make(map[string]struct{})
+	for _, name := range BidderNames() {
+		hashSet[string(name)] = struct{}{}
+	}
+	return hashSet
 }
 
-func (name *BidderName) String() string {
-	if name == nil {
-		return ""
+// bidderNameLookup is a map of the lower case version of the bidder name to the precise BidderName value.
+var bidderNameLookup = func() map[string]BidderName {
```

Comment:
Package private lookup for name normalization, since the adapter config can get an adapter name in lowercase but we're inconsistent and some adapter names are mixed case.